src/OrchardCore/OrchardCore.FileStorage.AzureBlob/BlobFileStore.cs

```diff
@@ -54,7 +54,7 @@ public BlobFileStore(BlobStorageOptions options, IClock clock, IContentTypeProvi
                 {
                     await _blobContainer.CreateIfNotExistsAsync();
                     await CreateBasePathIfNotExistsAsync();
-                    await _blobContainer.SetPermissionsAsync(new BlobContainerPermissions() { PublicAccess = BlobContainerPublicAccessType.Blob });
+                    await _blobContainer.SetPermissionsAsync(new BlobContainerPermissions() { PublicAccess = _options.PublicAccessType });
```

Comment:
Do you think we should create the container? That would also remove the public access config, and this weird task.run call that is cached. It can just be a requirement that a container exists. And the user defines its accessibility.

Reply:
I'm probably 75 / 25 on this.

It's normal behaviour to create the container, O1 did, ImageSharp does, as do a lot of other plugins.

Reasons for removing
- It's a bit of a weird call that has to get made. and it can fail due to some weirdness with Azure.Storage
- Responsibility should really be with the dev to configure the container, set the permissions how they want etc
- https://github.com/OrchardCMS/OrchardCore/issues/2394 An old issue of mine before the dog ate my github account. Essentially the easiest way to use a Shared Access Key, which I still think is a useful scenario for tenants (i.e. one blob account, but limit access to tenant containers w/sahred keys), is to not call `CreateIfNotExistsAsync`, just assume the blob exists, and allow Azure Storage to throw the exception if it doesn't

Reasons for keeping
- It's a little bit easier to setup because you don't have to create the container.

Reply:
75 > 25

Reply:
Done. 

Noting that the BasePath will get created automatically, on upload, but without the directory marker file. Which isn't necessary in the root 'directory' simulation, as an identifier, just in the child directories.